src/repositories/consent.ts

```diff
@@ -58,6 +78,53 @@ export class FirebaseConsentRepository implements IConsentRepository {
         })
     })
   }
+
+  async updateConsentById(
+    id: string,
+    data: Record<string, unknown>
+  ): Promise<void> {
+    await firebase.firestore().collection('consents').doc(id).update(data)
+  }
+
+  async updateConsent(
+    conditions: Record<string, unknown>,
+    data: Record<string, unknown>
+  ): Promise<void> {
+    let firestoreQuery: FirebaseFirestore.Query = firebase
+      .firestore()
+      .collection('consents')
+
+    // Chain all of the given conditions to the query
+    for (const key in conditions) {
+      firestoreQuery = firestoreQuery.where(key, '==', conditions[key])
+    }
+
+    // Find and update all matching documents in Firebase that match the given conditions.
+    firestoreQuery
+      .get()
+      .then((response) => {
+        // Create a batch to perform all of the updates using a single request.
+        // Firebase will also execute the updates atomically according to the
+        // API specification.
+        const batch = firebase.firestore().batch()
+
+        // Iterate through all matching documents add them to the processing batch.
+        response.docs.forEach((doc) => {
+          batch.update(
+            // Put a reference to the document.
+            firebase.firestore().collection('consents').doc(doc.id),
+            // Specify the updated fields and their new values.
+            data
+          )
+        })
+
+        // Commit the updates.
+        return batch.commit()
```

Comment:
if batch.commit() returns a Promise where it is resolved or where rejection is handled?

Reply:
good question. Actually I'm not completely sure where it should be handled. The endpoint handlers which call this method typically call it async and then send back a response - so probably shouldn't be handling there 
we could simply add a try catch or .then.catch - with the .then() returning void and .catch throwing/logging error?